spec.bs

```diff
@@ -759,6 +787,194 @@ in the [[#nested-traversables-intro]].
+    [=fenced navigable container/fenced navigable=], and |old chain| does not also [=list/contain=]
+    |document|, then return.
+
+    Note: This is how we bail-out early just before calling the [=focus update steps=], in the case
+    where focus is trying to cross the fence.
+</div>
+
+<h4 id=focusing-steps-callers>Changes to the callers of the [=focusing steps=] algorithm</h4>
+
+Modify the user agent sentence after the algorithm steps to read:
+
+User agents must [=immediately=] run the [=focusing steps=] for a [=focusable area=] or
+[=navigable=] |candidate| with [=focus-unfenced|unfenced=] set to true whenever the user attempts to
+move the focus to |candidate|.
+
+<div algorithm=access-key-patch>
+  Modify the action of the [=accesskey attribute command=] algorithm to be:
+
+  1. Run the [=focusing steps=] for the element with [=focus-unfenced|unfenced=] set to true.
+
+  1. <a>Fire a `click` event</a> at the element.
+</div>
+
+<div algorithm=activation-patch>
+  Modify the behavior when a user [=activation|activates=] a [=click focusable=] [=focusable area=]
+  to be:
+
+  When a user [=activation|activates=] a [=click focusable=] [=focusable area=], the user agent must
+  run the [=focusing steps=] on the [=focusable area=] with <var ignore>focus trigger</var> set
+  to "`click`" and [=focus-unfenced|unfenced=] set to true.
+</div>
+
+<div algorithm=hide-popover-patch>
+  Modify step 10 of the [=hide popover algorithm=] to read:
+
+  10. If |previouslyFocusedElement| is not null, then:
+      1. Set <var ignore>element</var>'s [=previously focused element=] to null.
+      2. If <var ignore>focusPreviousElement</var> is true, then run the [=focusing steps=] for
+        |previouslyFocusedElement| with [=focus-unfenced|unfenced=] set to true; the viewport
+        should not be scrolled by doing this step.
+
+  Note: Although dismissing a popover manually is a user-initiated gesture, the
+  [=focusing steps=] will be called with [=focus-unfenced|unfenced=] set to false regardless
+  of whether this was called from user gesture or via a script call.
+</div>
+
+<div algorithm=interactively-validate-patch>
+  Modify the first bullet point of step 3 of the [=interactively validate the constraints=]
+  algorithm to read:
+
+  * User agents may focus one of those elements in the process, by running the
+    [=focusing steps=] for that element, and may change the scrolling position of the
+    document, or perform some other action that brings the element to the user's attention.
+    If these steps were invoked by user gesture, [=focusing steps=] can be called with
+    [=focus-unfenced|unfenced=] set to true. For elements that are
+    [=form-associated custom elements=], user agents should use their [=face validation anchor=]
+    instead, for the purposes of these actions.
+</div>
+
+  <h4 id=has-focus-steps-changes>Changes to the [=has focus steps=] algorithm</h4>
```

Comment:
While these are somewhat useful, I think they end up polluting the sidebar contents, so may we can remove the `<h4>`s for each individual algorithm and instead just wrap all of these in `<div algorithm>`s to separate/identify them. Sound good?